### PR TITLE
Implement get and set methods of PublicAsset extension

### PR DESCRIPTION
**Pull request by tddang-linagora**
## Issue
- https://github.com/linagora/tmail-flutter/issues/2932
- https://github.com/linagora/tmail-flutter/issues/3040

### CODE REVIEW

File: test/jmap/mail/extensions/public_asset/get/get_public_asset_method_test.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      },
+      methodCallId.value
+    ]]
+  };
+  
+  group('get public asset method test:', () {
+    test(
+      'should return expected PublicAsset '
+      'when call method PublicAsset/get',
+    () async {
```
Comment (dab246) [inline test/jmap/mail/extensions/public_asset/get/get_public_asset_method_test.dart:69]: Also write test with case failed

File: test/jmap/mail/extensions/public_asset/get/get_public_asset_method_test.dart [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      // arrange
+      final method = GetPublicAssetMethod(accountId)..addIds({publicAsset.id!});
```
Comment (dab246) [inline test/jmap/mail/extensions/public_asset/get/get_public_asset_method_test.dart:71]:
```suggestion
      final getPublicAssetMethod = GetPublicAssetMethod(accountId)..addIds({publicAsset.id!});
```

File: test/jmap/mail/extensions/public_asset/set/set_public_asset_method_test.dart [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    contentType: 'image/jpeg',
+    publicURI: 'http://domain.com/public/abc123',
+    identityIds: {identityId: true}
+  );
+  
+  group('set public asset method test:', () {
+    test(
+      'should return created public asset '
+      'when PublicAsset/set create return success',
+    () async {
```
Comment (dab246) [inline test/jmap/mail/extensions/public_asset/set/set_public_asset_method_test.dart:40]: Please test more case failure

File: lib/jmap/mail/extensions/public_asset/public_asset.dart [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import 'package:jmap_dart_client/http/converter/identities/public_asset_identities_converter.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'public_asset.g.dart';
+
+typedef PublicAssetIdentities = Map<IdentityId, bool>;
+
+@JsonSerializable(
+  converters: [
+    IdConverter(),
```
Comment (hoangdat) [inline lib/jmap/mail/extensions/public_asset/public_asset.dart:14]:
```suggestion
    IdNullableConverter(),
```

File: lib/jmap/mail/extensions/public_asset/public_asset.dart [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import 'package:equatable/equatable.dart';
+import 'package:jmap_dart_client/http/converter/id_converter.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
+import 'package:jmap_dart_client/http/converter/identities/public_asset_identities_converter.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'public_asset.g.dart';
+
+typedef PublicAssetIdentities = Map<IdentityId, bool>;
```
Comment (hoangdat) [inline lib/jmap/mail/extensions/public_asset/public_asset.dart:10]: please test for this, also the failed cases in response body

File: lib/jmap/mail/extensions/public_asset/get/get_public_asset_method.dart [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import 'package:jmap_dart_client/http/converter/id_converter.dart';
+import 'package:jmap_dart_client/jmap/core/capability/capability_identifier.dart';
+import 'package:jmap_dart_client/jmap/core/method/method.dart';
+import 'package:jmap_dart_client/jmap/core/method/request/get_method.dart';
+import 'package:jmap_dart_client/jmap/core/request/request_invocation.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'get_public_asset_method.g.dart';
+
+@JsonSerializable(converters: [AccountIdConverter(), IdConverter()])
```
Comment (hoangdat) [inline lib/jmap/mail/extensions/public_asset/get/get_public_asset_method.dart:11]: how about to use nullable converter ?

Comment (tddang-linagora) [reply lib/jmap/mail/extensions/public_asset/get/get_public_asset_method.dart:11]:
```dart
mixin OptionalIds {
  @JsonKey(includeIfNull: false)
  Set<Id>? ids;

  void addIds(Set<Id> values) {
    if (ids == null) {
      ids = Set();
    }
    ids?.addAll(values);
  }
}
```

`Id` in `OptionalIds` is not null, so we cannot use `IdNullableConverter` there.

File: test/jmap/mail/extensions/public_asset/get/get_public_asset_method_test.dart (hunk 1 above)
Comment (tddang-linagora) [reply test/jmap/mail/extensions/public_asset/get/get_public_asset_method_test.dart:69]: Done

File: test/jmap/mail/extensions/public_asset/get/get_public_asset_method_test.dart (hunk 2 above)
Comment (tddang-linagora) [reply test/jmap/mail/extensions/public_asset/get/get_public_asset_method_test.dart:71]: Done

File: test/jmap/mail/extensions/public_asset/set/set_public_asset_method_test.dart (hunk 3 above)
Comment (tddang-linagora) [reply test/jmap/mail/extensions/public_asset/set/set_public_asset_method_test.dart:40]: Done

File: lib/jmap/mail/extensions/public_asset/public_asset.dart (hunk 4 above)
Comment (tddang-linagora) [reply lib/jmap/mail/extensions/public_asset/public_asset.dart:14]: Done

File: lib/jmap/mail/extensions/public_asset/public_asset.dart (hunk 5 above)
Comment (tddang-linagora) [reply lib/jmap/mail/extensions/public_asset/public_asset.dart:10]: Done

**Review by hoangdat — APPROVED** [review]

**Review by dab246 — APPROVED** [review]
- [ ] Please rebase

**Pull request merged**